### DISCUSSION TITLE
How can I get the variable written on the console in AWS codebuild buildspec

### DISCUSSION THREAD

**Discussion by sprabha1990** [GitTools/GitVersion#3125] [Q&A]
Hi,

My buildspec script runs the `dotnet-gitversion` tool to get the SemVer value of my dotnet6 package library. I was using the output type as buildserver which write the value in the console as expected. But once the process ends, I can't access the variable. Say the build phase in my buildspec have the following command 

```yaml
build
  commands
    - dotnet-gitversion /output buildserver
```

It writes all the variables and its values in the console as below:

```
Executing GenerateSetVersionMessage for 'CodeBuild'.
0.1.2-prerelease.23
Executing GenerateBuildLogOutput for 'CodeBuild'.
GitVersion_Major=0
GitVersion_Minor=1
GitVersion_Patch=2
GitVersion_PreReleaseTag=prerelease.23
GitVersion_PreReleaseTagWithDash=-prerelease.23
GitVersion_PreReleaseLabel=prerelease
GitVersion_PreReleaseLabelWithDash=-prerelease
GitVersion_PreReleaseNumber=23
GitVersion_WeightedPreReleaseNumber=23
GitVersion_BuildMetaData=
GitVersion_BuildMetaDataPadded=
GitVersion_FullBuildMetaData=Branch.stage.Sha.beb4f7f82dcb53bd4ba1c5c02750fd588e7fe38d
GitVersion_MajorMinorPatch=0.1.2
GitVersion_SemVer=0.1.2-prerelease.23
GitVersion_LegacySemVer=0.1.2-prerelease23
GitVersion_LegacySemVerPadded=0.1.2-prerelease0023
GitVersion_AssemblySemVer=0.1.2.0
GitVersion_AssemblySemFileVer=0.1.2.0
GitVersion_FullSemVer=0.1.2-prerelease.23
GitVersion_InformationalVersion=0.1.2-prerelease0023+Build.0.Date.20220524T043135.Branch.unknown.Sha.beb4f7f82dcb53bd4ba1c5c02750fd588e7fe38d
GitVersion_BranchName=stage
GitVersion_EscapedBranchName=stage
GitVersion_Sha=beb4f7f82dcb53bd4ba1c5c02750fd588e7fe38d
GitVersion_ShortSha=beb4f7f
GitVersion_NuGetVersionV2=0.1.2-prerelease0023
GitVersion_NuGetVersion=0.1.2-prerelease0023
GitVersion_NuGetPreReleaseTagV2=prerelease0023
GitVersion_NuGetPreReleaseTag=prerelease0023
GitVersion_VersionSourceSha=b17b9a5e36308ef8294a88dc5ac6a0aff01e71a3
GitVersion_CommitsSinceVersionSource=23
GitVersion_CommitsSinceVersionSourcePadded=0023
GitVersion_UncommittedChanges=0
GitVersion_CommitDate=20220524T043135
Outputting variables to 'gitversion.properties' ...
  INFO [05/24/22 16:45:29:02] Done writing
```

So, now I'm trying to get the value of `GitVersion_SemVer`in the post build phase for other operations like packing and pushing the nuget package.  But I'm not sure how to get it. Tried several combinations like `$GitVersion_SemVer`, `$(GitVersion_SemVer)`, etc.
  
Any help would be much appreciated.

Thanks!

**Comment by asbjornu** [comment]
From `CodeBuild.cs` it seems like the versoin variables are written to a file called `gitversion.properties`:

https://github.com/GitTools/GitVersion/blob/5564596a93531268cddb75425c652f2a42ad88a2/src/GitVersion.Core/BuildAgents/CodeBuild.cs#L13

https://github.com/GitTools/GitVersion/blob/5564596a93531268cddb75425c652f2a42ad88a2/src/GitVersion.Core/BuildAgents/CodeBuild.cs#L15

https://github.com/GitTools/GitVersion/blob/5564596a93531268cddb75425c652f2a42ad88a2/src/GitVersion.Core/BuildAgents/CodeBuild.cs#L33-L41

**Reply by sprabha1990** [reply]
Yes! I can able to read the variable from gitversion.properties file using "source" command. 

- dotnet-gitversion /output buildserver
 - source gitversion.properties

Now I can access SemVer using "$GitVersion_SemVer" in post build phase.

**Reply by asbjornu** [reply]
Excellent!
